Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.cpp

```diff
@@ -159,22 +193,25 @@ RefPtr<NativeImage> ImageBufferIOSurfaceBackend::copyNativeImage(BackingStoreCop
     return NativeImage::create(m_surface->createImage());
 }
 
-RefPtr<NativeImage> ImageBufferIOSurfaceBackend::copyNativeImageForDrawing(BackingStoreCopy)
+RefPtr<NativeImage> ImageBufferIOSurfaceBackend::copyNativeImageForDrawing(GraphicsContext& destination)
 {
-    return NativeImage::create(m_surface->createImage());
+    if (destination.hasPlatformContext() && CGContextGetType(destination.platformContext()) == kCGContextTypeBitmap) {
```

Comment:
The decision to call createIOSurfaceBitmapImageReference() is split in two places. ImageBuffer::draw() which checks whether `destination == context()` and this function which ensures we are drawing on unaccelerated context. Can't we have the decision in one place?

Reply:
No, these checks are for totally different purposes.
The check here is about "is the rendering deferred". "if so, we cannot return a reference".

The `destination == context()` is a different check. It is an existing check, totally unrelated to the code here, for "are we drawing onto the source." "if so, copy the source because otherwise it's not possible."

Reply:
This was also mentioned in slack that this hunk was overlapping with the other patch. So this is already merged via the other patch.